### PR TITLE
Added swagkeys transition lite keyboard

### PR DESCRIPTION
**Pull request by jiaxin96**
## Description

Added swagkeys transition lite

80% keyboard with RGB and VIA compatible

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/yandrstudio/transition_lite/readme.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# transition_lite
+
+![transition_lite](https://i.imgur.com/9IUQvsG.jpg)
+A 80% keyboard with RGB.
```
Comment (waffle87) [inline keyboards/yandrstudio/transition_lite/readme.md:4]:
```suggestion
![transition_lite](https://i.imgur.com/9IUQvsGh.jpg)

A 80% keyboard with RGB.
```

File: keyboards/yandrstudio/transition_lite/transition_lite.c [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      4,4,4,4,4, 4,4,4,4,4,
+      4,4,4,4,4, 4,4,4,4,4,
+      4,
+      8,8,8,8,8, 8,8,8,8,8,
+      8,8,8,8,8, 8,8,8,8,8,
+      8,8,8,8,8, 8
+
+    }
+};
+
```
Comment (waffle87) [inline keyboards/yandrstudio/transition_lite/transition_lite.c:42]: This should be migrated to `info.json` and removed from here.
```suggestion
```

File: keyboards/yandrstudio/transition_lite/transition_lite.c [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    }
+    /* Synchronize all led colors under the space key */
+    for (uint8_t i = 83; i <= 87; i++) {
+        rgb_matrix_ws2812_array[i].r = rgb_matrix_ws2812_array[85].r;
+        rgb_matrix_ws2812_array[i].g = rgb_matrix_ws2812_array[85].g;
+        rgb_matrix_ws2812_array[i].b = rgb_matrix_ws2812_array[85].b;
+    }
+
+    return true;
+}
```
Comment (waffle87) [inline keyboards/yandrstudio/transition_lite/transition_lite.c:57]:
```suggestion
#ifdef RGB_MATRIX_ENABLE
extern rgb_led_t rgb_matrix_ws2812_array[RGB_MATRIX_LED_COUNT];

bool rgb_matrix_indicators_advanced_kb(uint8_t led_min, uint8_t led_max) {
    if (!rgb_matrix_indicators_advanced_user(led_min, led_max)) {
        return false;
    }
    /* Synchronize all led colors under the space key */
    for (uint8_t i = 83; i <= 87; i++) {
        rgb_matrix_ws2812_array[i].r = rgb_matrix_ws2812_array[85].r;
        rgb_matrix_ws2812_array[i].g = rgb_matrix_ws2812_array[85].g;
        rgb_matrix_ws2812_array[i].b = rgb_matrix_ws2812_array[85].b;
    }

    return true;
}
#endif
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/yandrstudio/transition_lite/readme.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# transition_lite
+
+![transition_lite](https://i.imgur.com/9IUQvsG.jpg)
```
Comment (waffle87) [inline keyboards/yandrstudio/transition_lite/readme.md:3]:
```suggestion
![transition_lite](https://i.imgur.com/9IUQvsGh.jpg)
```

File: keyboards/yandrstudio/transition_lite/transition_lite.c [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
```
Comment (zvecr) [inline keyboards/yandrstudio/transition_lite/transition_lite.c:22]: This should probably avoid mutating the drivers array directly, and do something like the following
```suggestion
uint8_t rgb_matrix_map_row_column_to_led_kb(uint8_t row, uint8_t column, uint8_t *led_i) {
    if ( row == 5 && column == 7) {
        led_i[0] = 83;
        led_i[1] = 84;
        led_i[2] = 86;
        led_i[3] = 87;
        return 4;
    }
    return 0;
}
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Pull request closed without merging**